R/group_apply.R

```diff
@@ -0,0 +1,119 @@
+#' Apply a function over grouping(s) of data
+#'
+#' A function to apply `palaeoverse` functionality across subsets (groups) of
+#' data, delineated using one or more variables. `data.frame` related functions
+#' (e.g. `nrow`, `ncol`, `lengths`, `unique`) may also be used.
+#'
+#' @param occdf \code{dataframe}. A dataframe of fossil occurrences.
+#' This dataframe must contain the grouping variables and the necessary
+#' variables for the function you wish to call (see function specific
+#' documentation for required columns).
+#' @param group \code{character}. A vector of column names, specifying the
+#' grouping (e.g. "collection_no", "stage_bin").
+#' @param fun \code{function}. The function you wish to apply to
+#' `occdf`. See details for compatible functions.
+#' @param ... Additional arguments available in the called
+#' function. These additional arguments may be required for function arguments
+#' without default values, or if you wish to overwrite the default argument
+#' value (see examples).
+#'
+#' @return A \code{dataframe} of the outputs from the selected function, with an
+#' appended column indicating the user-defined grouping (`grouping`).
+#'
+#' @details `group_apply` enables the separate analysis of occurrences from
+#' different time intervals, spatial regions, or trait values. The function
+#' serves as a wrapper around `palaeoverse` functions on groupings of
+#' data. `data.frame` related functions (e.g. `nrow`, `ncol`, `lengths`,
+#' `unique`) may also be used.
+#' \cr
+#' \cr
+#' Relevant `palaeoverse` functions:
+#' - \code{\link{tax_unique}}: return the number of unique taxa per grouping
+#' variable.
+#' - \code{\link{tax_range_time}}: return the temporal range of taxa per
+#' grouping variable.
+#' - \code{\link{tax_range_geo}}: return the geographic range of taxa per
+#' grouping variable.
+#' - \code{\link{tax_check}}: return potential spelling variations of the
+#' same taxon per grouping variable. Note: `verbose` needs to be set to FALSE.
```

Comment:
If this is necessary, could add an 'if' switch to ensure this is the case?

Reply:
Hmmm I would prefer to update the function to set FALSE as default behaviour. I can imagine having issues with many if switches as we accumulate functions! Thoughts?

Reply:
Given that this would be the only switch at present, I'm not sure it really matters. Was there a reason behind setting the 'verbose' default to TRUE? That would be the only factor which would sway my opinion on this, I think.

Reply:
OK, let's leave it as it is. After checking tax_check, I think it is better to leave verbose set to TRUE purely because I don't see many users using tax_check with group_apply anyway...